[IMP] l10n_it_edi_sdicoop: added documentation

bfr-o opened the pull request.

jco-odoo commented:
@jcs-odoo Can you check what is still needed to merge this?

AntoineVDV commented on content/applications/finance/accounting/fiscal_localizations/localizations/italy.rst:
```diff
@@ -0,0 +1,66 @@
+Italy (Sdicoop)
+===============
+
+**Cette documentation est en cours de finalisation**
```
> Can you check what is still needed to merge this?

@jco-odoo Well, this? :p

bfr-o replied:
Our PO, @tsb-odoo would like for this documentation to be merged, even if it's not finalized yet. We are gathering informations to make sure that it is complete, but would like users to be able to consult it, but be warned that it might still change.

jcs-odoo commented on content/applications/finance/accounting/fiscal_localizations/localizations/italy.rst:
```diff
@@ -0,0 +1,66 @@
+Sending an invoice
+------------------
+
+First, make sure that FatturaPA is selected on your journal.
```
```suggestion
First, open your sales journal. Under the *Advanced Settings* tab, in the **Electronic invoicing**
field, check *FatturaPA* and *Save*.
```

jcs-odoo reviewed (changes requested):
Hello!

Here is my feedback.

If it's urgent, I can force push some changes, otherwise, everything is detailed in this message and in the comments.

Thank you :) 

- rewrite SDICoop with the right capitalization (instead of Sdicoop)
- [rst guidelines: break the line at 100th character](https://www.odoo.com/documentation/14.0/contributing/documentation/rst_guidelines.html#start-a-new-line-before-the-100th-character)
- [rst guideline: fix the indentation of your image directives](https://www.odoo.com/documentation/14.0/contributing/documentation/rst_guidelines.html#be-consistent-with-indentation)
- [Content guidelines: Add ALT tags to your images](https://www.odoo.com/documentation/14.0/contributing/documentation/content_guidelines.html#alt-tags)
- [content guidelines: rename your image files and put them in the appropriate folder](https://www.odoo.com/documentation/14.0/contributing/documentation/content_guidelines.html#media-files)
- [Compress your images with pngquant](https://www.odoo.com/documentation/14.0/contributing/documentation.html#pngquant)

bfr-o commented on content/applications/finance/accounting/fiscal_localizations/localizations/italy.rst:
```diff
@@ -0,0 +1,66 @@
+.. image:: media/italy_01.png
+  :align: center
+
+Note that all your file are encrypted upon reception in such a way that only you are able to decrypt it.
```
Multiple files.

bfr-o commented on content/applications/finance/accounting/fiscal_localizations/localizations/italy.rst:
```diff
@@ -0,0 +1,66 @@
+.. image:: media/italy_04.png
+  :align: center
+
+.. image:: media/italy_05.png
+  :align: center
```
It's to denote that the process takes place in two steps as people might expect it to go in one step. With the process shown, they will understand that it's ok if the invoice is not sent in one step.

I'm open to remove it if you still think it's unnecessary.

jcs-odoo replied:
Well, I think you pointed out where the info is. They'll see by themselves :)
The documentation isn't a tutorial video ^^

bfr-o commented on content/applications/finance/accounting/fiscal_localizations/localizations/italy_IT.rst:
```diff
@@ -1,6 +1,6 @@
-==========
-Italy (IT)
-==========
+=====================
+Italy (Pec mail) (IT)
+=====================
```
Not sure what is the best move here.

There is two italian e-invoicing system : SDICoop (web-service) that is the newly added one, in English and Pec mail (via e-mail) an old system that will eventually be removed, and the doc is in Italian.

Hence the two separate files. I think it's important to keep the different titles so that people are conscious that the two things are different. What is your opinion about file structure ?

jcs-odoo replied:
I would suggest doing this (+ adapt all the other headings)

```suggestion
==========
Italy (IT)
==========

.. _italy/pec:

PEC mail
========
```
And you can add a `:ref:` link in italy.rst in the warning at line 13.

As I said, ideally, all the content would be only in one single file in English (italy.rst) and the translation would be done in transifex.

bfr-o commented on content/applications/finance/accounting/fiscal_localizations/localizations/italy.rst:
```diff
@@ -0,0 +1,66 @@
+Italy (Sdicoop)
+===============
+
+**Cette documentation est en cours de finalisation**
```
I would keep it. @tsb-odoo  what is your opinion ?

bfr-o replied:
Translated it in English, though :D

jcs-odoo replied:
No. Even if it is the case, we don't write such kind of content. It is too often forgotten and left unmaintained.
Whenever we improve the content... well the page will be better, but no need to warn the readers.

bfr-o commented:
@jcs-odoo Thanks for your review ! I corrected a lot of things based on your input. Also, I added comments on some of your comments for the things that are still open for discussion. Let me know if that's ok.

jcs-odoo commented on content/applications/finance/accounting/fiscal_localizations/localizations/italy.rst:
```diff
@@ -0,0 +1,84 @@
+This module allows the creation of the EDI documents and the communication with the SDICoop
+web-service of FatturaPA to send/receive invoices and notifications.
+
+.. warning::
+  Once this module is installed, it's no longer possible to send invoice via PEC
+  mail.
```
In our RST guidelines, the indentation is aligned with the directive. So usually 3 white spaces, same for the image options.
https://www.odoo.com/documentation/14.0/contributing/documentation/rst_guidelines.html#be-consistent-with-indentation

Also, you break the line only once you reach the 100th character

+ invoices plural (as in my previous comment)

```suggestion
.. warning::
   Once this module is installed, it's no longer possible to send invoices via PEC mail.
```

jcs-odoo commented on content/applications/finance/accounting/fiscal_localizations/localizations/italy.rst:
```diff
@@ -0,0 +1,84 @@
+
+Allow the user to generate the EDI document for Italian invoicing.
+
+This module allows the creation of the EDI documents and the communication with the SDICoop
+web-service of FatturaPA to send/receive invoices and notifications.
```
```suggestion
This module allows the creation of the EDI documents and the communication with the SDICoop web
service of FatturaPA to send or receive invoices and notifications.
```

jcs-odoo commented on content/applications/finance/accounting/fiscal_localizations/localizations/italy.rst:
```diff
@@ -0,0 +1,84 @@
+   dell’indirizzo telematico dove ricevere tutte le fatture elettroniche`, input Odoo's Codice
+   Destinatario (**K95IV18**), then confirm.
+
+Give Odoo permission to process file
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
plural files
```suggestion
Give Odoo permission to process files
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

jcs-odoo commented on content/applications/finance/accounting/fiscal_localizations/localizations/italy.rst:
```diff
@@ -0,0 +1,84 @@
+  :alt: Register
+
+.. Note::
+  All your files are encrypted upon reception in such a way that only you are able to decrypt
+  it.
```
```suggestion
   All your files are encrypted upon reception in such a way that only you are able to decrypt them.
```
- indentation
- "them" (previous comment)
- no line break necessary as you haven't reached the 100th character yet.

jcs-odoo commented on content/applications/finance/accounting/fiscal_localizations/localizations/italy.rst:
```diff
@@ -0,0 +1,84 @@
+Enable FatturaPA on the Sales journal
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Open your sales journal. Under the *Advanced Settings* tab, in the **Electronic invoicing**
+field, check *FatturaPA* and *Save*.
```
```suggestion
Open your sales journal. Under the *Advanced Settings* tab, in the **Electronic invoicing** field,
check *FatturaPA* and *Save*.
```
line break at 100th

jcs-odoo commented on content/applications/finance/accounting/fiscal_localizations/localizations/italy.rst:
```diff
@@ -0,0 +1,84 @@
+To do this, go to :menuselection:`Accounting --> Settings --> Electronic Invoicing` and click on 
+*Register*.
+
+.. image:: media/italy-register.png
+  :align: center
+  :alt: Register
```
- the new media files should be in their own folder "italy", which also means the filename doesn't need the "italy" prefix anymore. It is detailed here:   https://www.odoo.com/documentation/14.0/contributing/documentation/content_guidelines.html#media-files
- indentation
- alt tag isn't good (see suggestion) https://www.odoo.com/documentation/14.0/contributing/documentation/content_guidelines.html#alt-tags

 ```suggestion
.. image:: italy/register.png
   :align: center
   :alt: Register button to allow Odoo to process invoices.
```

Please adapt the other images as well, thanks :)

bfr-o replied:
You point out indentation, I added a space (3 spaces) to ':align' and ':alt' so that it matches the indentation of image. However, I've seen example in other .rst where there are only 2 spaces but they are older files. Can you double check ?

jcs-odoo replied:
Indeed, it's a "new" rule and there are still way too many old docs ^^ I'll check yes

jcs-odoo reviewed (changes requested):
Here you go :)

jcs-odoo commented:
Additional question: is this only from v14 and up, or is it also valid for v13?

robodoo commented:
[Pull request status dashboard](https://mergebot.odoo.com/odoo/documentation/pull/1041).

robodoo commented:
Sorry, I didn't know about this PR and had to retrieve its information, you may have to re-approve it.

bfr-o commented:
@jcs-odoo adapted according to your comments. Just one last question here: https://github.com/odoo/documentation/pull/1041#discussion_r678291403

To answer your question: the feature is only available in v14 and up.

Thanks.

jcs-odoo reviewed (approved):
Looking good. Good job :)

I left some last suggestions but it's already "pre-approved"

I also think the ALT tags could be better but it isn't a blocking issue.
Please remove the warning "still a work in progress" in bold at the beginning.

Thank you for the doc and have a good evening!

jcs-odoo commented:
@tsb-odoo last changes force pushed

AntoineVDV commented:
@robodoo r+

robodoo commented:
I'm sorry, @AntoineVDV. I'm afraid I can't do that.

AntoineVDV commented:
@robodoo r+
